Module: Lich Pet Summon fix

Argavyon opened the pull request:
Makes the overly wide petlist for liches with summon zombie+skeleton+FM+necro more eye-friendly.

Goes from this
![Screenshot_3](https://user-images.githubusercontent.com/75849479/153099857-9e86fdeb-fee2-4467-af87-6da6062259fb.jpg)
To this
![Screenshot_4](https://user-images.githubusercontent.com/75849479/153099870-ea38ab3c-5334-4553-8b35-08159be3405c.jpg)

Argavyon committed:
Module: Lich Pet Summon fix
Makes the overly wide petlist for liches with summon zombie+skeleton+FM+necro more eye-friendly.

Argavyon committed:
minute style changes

AnneTrue commented on nexus-tweaks.user.js:
```diff
@@ -1144,6 +1144,43 @@ promiseList.push((async () => {
+
+    // Pets summon buttons
+    const petSummons = petTableTable.querySelector('tbody').querySelector('table.summonsButtons>tbody');
+    const FM = petSummons.querySelector('input[value="Fossil Monstrosity"]');
```
Style-wise it's consistent as "fossilMonstrosity", rather than all-caps.

AnneTrue commented on nexus-tweaks.user.js:
```diff
@@ -1144,6 +1144,43 @@ promiseList.push((async () => {
+    const necro = petSummons.querySelector('input[value="Necrophage"]');
+
+    if (FM) {
+      FM.nextSibling.value = 'Fossil Monstrosity (3 Skeletons) '
```
What's the purpose of this? Are you cutting off the MP cost?

AnneTrue commented on nexus-tweaks.user.js:
```diff
@@ -1144,6 +1144,43 @@ promiseList.push((async () => {
+    if (FM) {
+      FM.nextSibling.value = 'Fossil Monstrosity (3 Skeletons) '
+      if (petSummons.children.length < 3) petSummons.appendChild(document.createElement('tr'));
+      petSummons.children[2].insertBefore(FM.parentNode.parentNode, petSummons.children[2].firstChild);
```
When you're manipulating DOM elements with relative positions like this, it helps a lot to document what you expect those elements to be.

AnneTrue reviewed (changes requested):
Two overall questions:
- Are the devs fixing this directly? This is very much a problem they should fix.
- Is this problem actually specific to liches because they have the dual titans, or will other classes benefit from a more generic approach?

Besides that, needs the usual check-off of pre-merge items such as README documentation and version bumps.

Argavyon committed:
fix

Corrected the display for skeleton+zombie+necro (necro would display on the first row instead of the second one)

Argavyon committed:
style fixes

FM -> fossilMonstrosity
commented on DOM modifications

Argavyon committed:
Lich Pet Summon fix documentation

Argavyon commented:
Version bumping isn't trivial from my side, as I can't tell what will be merged first. (In particular, whether this will be merged before the badge classifier).
This would be an even larger issue if someone else was sending PRs.

Argavyon committed:
Merge branch 'main' into Lich-Pet-Summoning-Fix
